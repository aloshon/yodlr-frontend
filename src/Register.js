import React from "react";
import {useHistory} from "react-router-dom";
import useFields from "./useFields";
import Container from "react-bootstrap/Container";
import Form from "react-bootstrap/Form";
import Button from "react-bootstrap/Button";
import axios from "axios";

/**
 * Register renders form for users to register their info
 * User cannot input blank or "space-only" inputs
 * If user is already registered, the user will be signed in
 * Redirect to homepage after registering
 */
const Register = () => {
    const history = useHistory();
    // id is only there so new data is stored in the same order
    const [formData, handleChange, resetFormData] = useFields({
        id: "",
        email: "",
        firstName: "",
        lastName: ""
    });

    const handleSubmit = async (e) => {
        e.preventDefault();
        // users are sent to homepage right after so set them active
        const {data} = await axios.post("http://localhost:3001/users", 
        {...formData, state: "active"});
        
        localStorage.setItem("user", JSON.stringify(data));
        history.push("/homepage");
    }

    // remove all spaces from every input,
    // if there is actual data (not just spaces) in all, return true
    // Once all are not empty then allow user to register
    let allFilled = formData.email.replace(/ /g, '') !== "" &&
        formData.firstName.replace(/ /g, '') !== "" &&
        formData.lastName.replace(/ /g, '') !== ""
    
    return (
        <Container>
            <h1 style={{marginBottom: "100px", fontSize: "60px"}}>Register</h1>
            <Form onSubmit={handleSubmit}>
                <Form.Group>
                    <Form.Label id="firstName">First Name</Form.Label>
                    <Form.Control 
                        name="firstName" 
                        type="text" 
                        aria-labelledby="firstName"
                        minLength={1}
                        value={formData.firstName}
                        onChange={handleChange}
                        placeholder="first name"
                        />
                </Form.Group>
                <Form.Group>
                    <Form.Label id="lastName">Last Name</Form.Label>
                    <Form.Control 
                        name="lastName" 
                        type="text" 
                        aria-labelledby="lastName"
                        minLength={1}
                        value={formData.lastName}
                        onChange={handleChange}
                        placeholder="last name"
                        />
                </Form.Group>
                <Form.Group>
                    <Form.Label id="email">Email</Form.Label>
                    <Form.Control 
                        name="email" 
                        type="email"
                        aria-labelledby="email"
                        minLength={5}
                        value={formData.email}
                        onChange={handleChange}
                        placeholder="email"
                        />
                </Form.Group>
                <Button disabled={!allFilled} variant="primary" type="submit" size="lg" block>
                    Submit
                </Button>
            </Form>
        </Container>
    )
}

export default Register;