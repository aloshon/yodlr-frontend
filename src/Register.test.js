import { render } from '@testing-library/react';
import Register from './Register';
import {Router} from 'react-router-dom';
import { createMemoryHistory } from 'history'

describe("Register should render", () => {
    const history = createMemoryHistory();
    const {asFragment, getByLabelText, getByText} = render(
        <Router history={history}>
            <Register />
        </Router>
    );


    test('Form for user registration is loaded', () => {
        const email = getByLabelText("Email");
        const first = getByLabelText("First Name");
        const last = getByLabelText("Last Name");
        const submit = getByText("Submit");
        expect(email).toBeInTheDocument();
        expect(first).toBeInTheDocument();
        expect(last).toBeInTheDocument();
        expect(submit).toBeInTheDocument();
    });
    
    test("matches snapshot", () => {
        expect(asFragment()).toMatchSnapshot();
    });
});