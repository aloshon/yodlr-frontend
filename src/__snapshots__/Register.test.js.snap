// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register should render matches snapshot 1`] = `<DocumentFragment />`;
